lib/ec2/host/ec2_client.rb

```diff
@@ -37,11 +38,31 @@ def ec2
       def credentials
         if Config.aws_access_key_id and Config.aws_secret_access_key
           Aws::Credentials.new(Config.aws_access_key_id, Config.aws_secret_access_key)
-        else
-          Aws::SharedCredentials.new(profile_name: Config.aws_profile, path: Config.aws_credential_file)
+        else # use profile
+          shared_credentials = Aws::SharedCredentials.new(
+            profile_name: Config.aws_profile,
+            path: Config.aws_credential_file
+          )
+          config = Aws.shared_config.instance_variable_get(:@parsed_config)[Config.aws_profile]
```

Comment:
I dislike to use `instance_variable_get`.
I investigated about AWS SDK, but it looed AWS::SharedConfig class itself is also private https://github.com/aws/aws-sdk-ruby/blob/7c9584bc6473100df9aec9333ab491ad4faeeca8/gems/aws-sdk-core/lib/aws-sdk-core/shared_config.rb#L3-L4.

Referring codenize-tools https://github.com/codenize-tools/piculet/pull/28/files, it was using https://github.com/a2ikm/aws_config gem. Do you think it is good idea to use this gem (depend on this gem)?

Reply:
> https://github.com/codenize-tools/piculet/pull/28/files

This is good example!
i'll try it.